app/src/main/java/fr/free/nrw/commons/media/MediaDetailPagerFragment.java

```diff
@@ -332,13 +320,31 @@ private void updateBookmarkState(MenuItem item) {
 
     public void showImage(int i, boolean isWikipediaButtonDisplayed) {
         this.isWikipediaButtonDisplayed = isWikipediaButtonDisplayed;
-        Handler handler =  new Handler();
-        handler.postDelayed(() -> pager.setCurrentItem(i), 5);
+        settingViewPagerCurrentItem(i);
     }
 
     public void showImage(int i) {
-        Handler handler =  new Handler();
-        handler.postDelayed(() -> pager.setCurrentItem(i), 5);
+        settingViewPagerCurrentItem(i);
+    }
+
+  /**
+   * This function waits for the item to load then sets the item to current item
+   * @param position current item that to be shown
+   */
+    private void settingViewPagerCurrentItem(int position) {
```

Comment:
I just noticed the method name. As you probably know, Java method names should begin with a verb, except for some rare exceptions. Do you have a special reason for not using a verb? If not, would you mind renaming it? Thanks!

Reply:
No, there is no specific reason for that. Even thought I always use verb, i wasn't aware about this hard and fast rule. Thanks for telling me.